server/consumer.go

```diff
@@ -5162,7 +5162,7 @@ func (o *consumer) stopWithFlags(dflag, sdflag, doSignal, advisory bool) error {
 			acc.mu.RUnlock()
 		}
 		if jsa != nil {
-			isConsumerAssigned = jsa.consumerAssigned(stream, consumer)
+			consumerDeleted = !jsa.consumerAssigned(stream, consumer) || !jsa.js.isClustered()
```

Comment:
I suggest what I recommended, since it will be cleared that consumerDeleted is always true when not clustered. This version is not as clear IMO.

Reply:
I went with this approach, as `assignment` is a clustered JetStream concept, and what we do care here is if the consumer should be deleted or not.

Anyway, I updated the PR as you suggested.